css_tasks/task10/js/form.js

```diff
@@ -0,0 +1,157 @@
+import {constants} from './constants.js';
+import {DOMElements} from './elements.js';
+
+const {
+  nameText,
+  namePattern,
+  emailText,
+  emailPattern,
+  passwordPattern,
+  passwordText,
+  successText,
+} = constants;
+const {
+  formInputContainer,
+  inputs,
+  signupForm,
+  formButton,
+  formText,
+} = DOMElements;
+
+const submitForm = (e) => {
+  e.preventDefault();
+  validateName(e.target);
+  validateEmail(e.target);
+  validatePassword(e.target);
+  showSuccessMessage();
+};
+
+const validateName = (formName) => {
+  const name = formName.elements.name;
+  const nameValue = name.value;
+  hideWarning(name);
+  name.addEventListener('focus', () => removeDisabledState(formButton));
+  if (!namePattern.test(nameValue)) {
+    handleError(name, nameText, formButton);
+  }
+};
+
+//  Validate Email Input
+
+const validateEmail = (formName) => {
+  const email = formName.elements.email;
+  const emailValue = email.value;
+  hideWarning(email);
+  email.addEventListener('focus', () => removeDisabledState(formButton));
+  if (!emailPattern.test(emailValue)) {
+    handleError(email, emailText, formButton);
+  }
+};
+
+// Validate Password Input
+
+const validatePassword = (formName) => {
```

Comment:
all 3 validation functions can be turned into 1 that takes a validation callback of any form of delegation.